include/soci/column-info.h

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright (C) 2016 Maciej Sobczak
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef SOCI_COLUMN_INFO_H_INCLUDED
+#define SOCI_COLUMN_INFO_H_INCLUDED
+
+#include "soci/soci-backend.h"
+#include "soci/type-conversion.h"
+#include "soci/values.h"
+
+namespace soci
+{
+
+struct SOCI_DECL column_info
+{
+    std::string name;
+    data_type type;
+    std::size_t length; // meaningful for text columns only
+    std::size_t precision;
+    std::size_t scale;
+    bool nullable;
+};
+
+std::size_t get_numeric_value(const values & v,
```

Comment:
Is this going to be a public API? If not, it should be in the `details` sub-namespace.